packages/flutter_driver/lib/src/driver/web_driver.dart

```diff
@@ -294,6 +294,29 @@ class FlutterWebConnection {
 
   /// Sends command via WebDriver to Flutter web application.
   Future<dynamic> sendCommand(String script, Duration? duration) async {
+    // This code should not be reachable before the VM service extension is
+    // initialized. The VM service extension is expected to initialize both
+    // `$flutterDriverResult` and `$flutterDriver` variables before attempting
+    // to send commands. This part checks that `$flutterDriverResult` is present.
+    // `$flutterDriver` is not checked because it is covered by the `script`
+    // that's executed next.
+    try {
+      await _driver.execute(r'return $flutterDriverResult', <String>[]);
```

Comment:
Have you considered wrapping this `await` in a `waitFor` call, similar to here:

https://github.com/flutter/flutter/blob/549f70c9146c5286f4aeca163724b4fbb402eec0/packages/flutter_driver/lib/src/driver/web_driver.dart#L304-L308

(I'd suggest a much shorter duration than 30 days, though)

Just to have some sort of watchdog that the await is not going to hang "forever" if it never ends up throwing?

Reply:
This is effectively removing a timeout, which is something we've been doing across the entire codebase. We already have global test timeouts that should be sufficient. If it hangs, we'll know there a bug and fix it.

Reply:
The waitFor I suggested is not the one from flutter_driver, but from `'package:webdriver/support/async.dart'`